File: Algorithms/Searching & Sorting/Bubble Sort/BubbleSort.py
def bubble_sort(a):
    n = len(a)

    for i in range(n):
        # Last i elements are already in place
        for j in range(0, n - i - 1):
            # Traverse the array from 0 to n-i-1
            # Swap if the element found is greater than the next element
            if a[j] > a[j + 1]:
                a[j], a[j + 1] = a[j + 1], a[j]


if __name__ == '__main__':
    a = [64, 34, 25, 12, 22, 11, 90]
    bubble_sort(a)
    print("Sorted array is:")
    for i in range(len(a)):
        print("%d" % a[i])

File: Algorithms/Searching & Sorting/Bubble Sort/bubble_sort.cs
using System;

public class BubbleSort
{

	public int[] bubblesort(int[] array) {
		while(true) {
			int swap_count = 0;
			for(int i=0; i<array.Length-1; i++) {
				if(array[i]>array[i+1]) {
					int swap = array[i];
					array[i] = array[i+1];
					array[i+1] = swap;
					swap_count += 1;
				}
			}
			if(swap_count == 0) {
				break;
			}
		}
		return array;
	}

	public static void Main(string[] args)
	{
		int[] arr = new int[6]{1, 6, 2, 8, 2, 3};
		arr = new BubbleSort().bubblesort(arr);
		Console.WriteLine("Sorted array is");
		for(int i=0;i<arr.Length;i++) {
			Console.Write(arr[i] + " ");
		}
	}
}
